[nnc] Test case for computeAt with reduction

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #54756 [nnc] Disabled test case for failure in implementing conv1d
* **#54755 [nnc] Test case for computeAt with reduction**

As title.  A step on the way to using computeAt to optimize
convolution.

Differential Revision: [D27353663](https://our.internmc.facebook.com/intern/diff/D27353663/)

bertmaher committed:
[nnc] Test case for computeAt with reduction

As title.  A step on the way to using computeAt to optimize
convolution.

Differential Revision: [D27353663](https://our.internmc.facebook.com/intern/diff/D27353663/)

[ghstack-poisoned]

ZolotukhinM commented on test/cpp/tensorexpr/test_loopnest.cpp:
```diff
@@ -1892,6 +1892,136 @@ TEST(LoopNest, LoopNestComputeAt_3) {
 
+using Axis = const VarHandle&;
+
+void checkIR(Stmt* s, const std::string& pattern) {
+  std::ostringstream oss;
+  oss << *s;
+  torch::jit::testing::FileCheck().run(pattern, oss.str());
+}
```
Nit: since you're adding this, it would be good to use it in existing tests too.

bertmaher replied:
Yep, will update things in a follow up diff :-)

ZolotukhinM commented on test/cpp/tensorexpr/test_loopnest.cpp:
```diff
@@ -1892,6 +1892,136 @@ TEST(LoopNest, LoopNestComputeAt_3) {
+    l.simplify();
+    checkIR(l.root_stmt(), R"IR(
+# CHECK: for (int py = 0; py < H + 1; py++) {
+# CHECK:   for (int px = 0; px < W + 1; px++) {
+# CHECK:     Allocate(prod); // dtype=int, dims=[H + 1, W + 1]
+# CHECK:     prod[(px + py * W) + 1] = px * py;
+# CHECK:     Free(prod);
+# CHECK:   }
+# CHECK: }
```
I think you could nuke this loopnest if you add `l.eliminateDeadStores()`

bertmaher replied:
Woah I did not know eliminateDeadStores was a thing.  Will check it out.

ZolotukhinM reviewed (approved):
It works, nice! :)

bertmaher committed:
Add eliminateDeadStores on "[nnc] Test case for computeAt with reduction"


As title.  A step on the way to using computeAt to optimize
convolution.

Differential Revision: [D27353663](https://our.internmc.facebook.com/intern/diff/D27353663/)

[ghstack-poisoned]

bertmaher committed:
Add eliminateDeadStores on "[nnc] Test case for computeAt with reduction"


As title.  A step on the way to using computeAt to optimize
convolution.

Differential Revision: [D27353663](https://our.internmc.facebook.com/intern/diff/D27353663/)

[ghstack-poisoned]
